src/app/models/usermodel.js

```diff
@@ -68,14 +68,20 @@ const accountsModel = {
         return fetch(accountsUrl, {credentials: 'include'})
             .then(
                 (response) => {
-                    return response.json().then((result) => {
-                        if (window.dataLayer) {
-                            window.dataLayer.push({
-                                faculty_status: result.faculty_status
-                            });
+                    return response.json().then(
+                        (result) => {
+                            if (window.dataLayer) {
+                                window.dataLayer.push({
+                                    faculty_status: result.faculty_status
+                                });
+                            }
+                            return result;
+                        },
+                        (err) => {
+                            console.warn('No JSON in Accounts response');
+                            return {err};
                         }
-                        return result;
-                    });
+                    );
```

Comment:
interesting, this is essentially what kinetic does as well.  I wonder how hard it would be to expose the user's json as a `window.user` global or something so kinetic and other pages JS could use it.  It's not clear if this is fetched on every page though

Reply:
I created a UserContext that wraps the main Router code. That calls this; almost all user-checking code uses the Context. I should check that the few things that still do use it directly need to.